### PR TITLE
Update to 12.15 and build with OpenSSL 3 [skip ci]

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
Update to 12.15 (which is the latest in the 12 series) and build with OpenSSL 3. 12.9 was the first version to be compatible with OpenSSL 3. Bumping to 12.15 shouldn't have any effect on anything.

I also removed and unused patch and fixed linting issues.

### CODE REVIEW

File: abs.yaml [hunk 1]
```diff
@@ -2,3 +2,9 @@
 build_parameters:
   - "--suppress-variables"
   - "--skip-existing"
+
+channels:
```
Comment (danpetry) [inline abs.yaml:6]: will versions in this channel be overwritten ? i.e. will new commits after review result in updated packages or will it `--skip-existing`?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -51,22 +51,20 @@ outputs:
...
         - {{ pin_subpackage('libpq', exact=True) }}
     test:
-      requires:
-        - conda-build
       commands:
         - postgres --help
         - conda inspect linkages postgresql  # [not win]
         - conda inspect objects postgresql  # [osx]
-    script: install_db.sh   # [unix
+    script: install_db.sh   # [unix]
```
Comment (danpetry) [inline recipe/meta.yaml:67]: usually this field goes just after `name:`, would be better moved up there

File: recipe/meta.yaml [hunk 3]
```diff
@@ -51,22 +51,20 @@ outputs:
         - make             # [unix]
       host:
         # these are here for sake of run_exports taking effect
-        - krb5
-        - openssl
+        - krb5 1.20.1
+        - openssl {{ openssl }}
         - readline         # [not win]
```
Comment (danpetry) [inline recipe/meta.yaml:56]: is there a reason why you didn't pin the other host entries?

File: recipe/meta.yaml (hunk 2 above)
Comment (danpetry) [reply recipe/meta.yaml:67]: both here and below

File: abs.yaml (hunk 1 above)
Comment (JeanChristopheMorinPerso) [reply abs.yaml:6]: Right now no, because `--skip-existing` is set, at least for this package.

Comment (danpetry) [reply abs.yaml:6]: yeah sorry, to be clearer, you should remove skip-existing if you want the versions to be the ones corresponding to the last build on this PR - i.e., after everything has been reviewed

Comment (ryanskeith) [reply abs.yaml:6]: This is fine as the version changed as well.

**Review by ryanskeith — APPROVED** [review]
LGTM

File: abs.yaml (hunk 1 above)
Comment (danpetry) [reply abs.yaml:6]: ah you are going to remove the channels: stuff before merge and take the artifacts uploaded in the merge commit? I was thinking you would take the artifacts from anaconda.org and copy them across.

**Comment by danpetry** [comment]
@JeanChristopheMorinPerso just wondering if you have any thoughts about the comments.

**Comment by JeanChristopheMorinPerso** [comment]
@danpetry Sorry, this felt through the cracks. I addressed your comments.

**Review by danpetry — APPROVED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
Taking a look.

**Comment by danpetry** [comment]
great, thanks

**Pull request merged**